### ISSUE TITLE
Ahoy! Live Chat 

### ISSUE THREAD

**Issue by RosieBellini** [marcusjrc/lbuilder#24]
Hey guys, me and Marcus are currently doing some testing - can you alert us here if there are any major changes to the code so I can reflect it in the group report.

PS. we're definitely going to break the code, just you wait

**Comment by tpict** [comment]
we may or may not need a new class to fix #23

**Comment by tpict** [comment]
just pushed some changes to the equals() methods for Coordinate and PathNode (though this should make them easier to test)

**Comment by RosieBellini** [comment]
Okay, currently on blackbox testing against the spec we've been given so for the time being I don't think that'll affect me.

Could also just do with some brief javadoc comments as we're coming across methods we don't really know what they're supposed to do? (makes testing a wee bit harder)

**Comment by tpict** [comment]
spritemap and savestate are fully documented if you want to have a look at them - tackled those first since they're (more or less) complete. i'll do docs where i can but just ask if there's anything you don't understand

**Comment by RosieBellini** [comment]
Okay thanks. I do now have an issue - can't seem to open the main game for the screenshots required for black box testing as it's chucking up an exception in main

**Comment by tpict** [comment]
what does it say?

**Comment by RosieBellini** [comment]
Exception in thread "main" java.lang.Error: Unresolved compilation problems: 
	The constructor PathNode(Coordinate, Coordinate, Coordinate) is undefined
	The constructor PathNode(Coordinate, Coordinate, Coordinate) is undefined
	The constructor PathNode(Coordinate, Coordinate, Coordinate) is undefined
	The constructor PathNode(Coordinate, Coordinate, Coordinate) is undefined

	at team1.PathNodeTest.main(PathNodeTest.java:16)

**Comment by RosieBellini** [comment]
Thanking you xx

**Comment by tpict** [comment]
fixed in 67d96b8fb8846e9eed55c6dde56440391143c8e2

**Comment by marcusjrc** [comment]
Just writing a test class for the Coordinate class. Is there any reason both x and y are public variables? it's just I thought using accessor methods was a better way to ensure encapsulation and whatnot.

**Comment by tpict** [comment]
i modelled it after the Dimension class which does the same thing, if you try to change them from anywhere you'll get a compilation error one way or the other. also make sure you pull the latest version of Coordinate

**Comment by tpict** [comment]
also thoughts on #18 ?

**Comment by RosieBellini** [comment]
Personal favourite combination of game so far: drum scoundrel

**Comment by marcusjrc** [comment]
What was it you guys were saying about Coordinates been allowed negative numbers again?

**Comment by tpict** [comment]
Coordinates can be negative, but the game map exists in the positive quadrant only. you can still place things in negative positions though, they just won't be displayed (useful as you can, for example, have no player on the screen, but still obey the "one player only" rule)

**Comment by tpict** [comment]
you guys will want to check out b7d087f5a35c1fc09d84e54b052cf8e7dc119fd9

**Comment by tpict** [comment]
check #18 again, i've removed FixedSizeStack

**Comment by tpict** [comment]
19f49554130968eb0ca1abcedde89a04139d8e45 has a lot of changes in it, have a look

**Comment by RosieBellini** [comment]
Fab, we've finished for the day but we'll be back for more.
Black box testing has been completed with a report to go with and Marcus is about half way through white box testing. Thanks guys xx

**Comment by RosieBellini** [comment]
Yo just remembered, shouldn't we give credit to the people who've made our levels? Could put a quick note in the 'helper' section?

**Comment by tpict** [comment]
yeah i'll make sure to credit lee harvey oswald in the about dialogue

**Comment by RosieBellini** [comment]
Round uuup. Is there anything the team should be aware of before we submit tomorrow?

**Comment by tpict** [comment]
need josh to write docs for the solver but otherwise i think we're good. please keep playing & testing though

**Comment by RosieBellini** [comment]
Any major re-factoring since we've been away?

**Comment by RosieBellini** [comment]
You guys have automated it! Beautiful :dancer:

**Comment by tpict** [comment]
you didn't have that when you were testing? check the commit log because some stuff got moved around

**Comment by RosieBellini** [comment]
Not whilst we were testing, we were testing whichever version was active from 2:30 yesterday

**Comment by tpict** [comment]
i pushed the pathfinder code at 2PM, it's in SokobanMap

**Comment by RosieBellini** [comment]
Marcus has made test sets of a few of the classes on the new branch

**Comment by tpict** [comment]
ah i didn't see that, we'll have to merge it with master at some point (i assume they want test code when we submit the source)

**Comment by RosieBellini** [comment]
Half the main classes have testcases, currently working on the others. I've done a document we can hand in named "Evidence of Testing" as it's impossible to demonstrate black box testing on the code

**Comment by tpict** [comment]
formal documentation for black box testing on a video game seems impossible for sure. but yeah just make sure you're testing the most recent commit, afaik there's nothing to change until bugs are found

**Comment by RosieBellini** [comment]
I've tried to break the GUI, only issue I can find is when the game is clicked off, the solver window disappears but runs in the background. Rating of minor issue to no issue at all.

**Comment by tpict** [comment]
is that running the compiled .jar or running the code in eclipse? because i don't have that problem... let me try windows real quick

**Comment by RosieBellini** [comment]
Running the code in eclipse, I'll try the .jar

**Comment by tpict** [comment]
i don't have that issue in windows either so hopefully it's just an eclipse thing

**Comment by userzjw** [comment]
I just tried myself both jar and eclipse, i didn't find that issue as well

**Comment by RosieBellini** [comment]
If you try the solver on the harder levels where it takes more time, click away from the game and then back onto it, on both the .jar file and the eclipse version the solver window disappears

**Comment by tpict** [comment]
what version of windows are you using?

**Comment by userzjw** [comment]
Maybe the solver window is behind the main interface? I just did run the solver and click away, the solver is hidden behind the main interface, i move the main interface a little and i saw the solver window

**Comment by tpict** [comment]
yeah i think josh mentioned that happening on windows, but i'm not seeing it on windows 10

**Comment by userzjw** [comment]
i just find that issue as well on windows 7

**Comment by RosieBellini** [comment]
Windows 7 - probably nothing to concern yourself with.

**Comment by tpict** [comment]
the machines on campus are running 8 right? hopefully it doesn't happen on those. it would be kind of annoying if that happened in the demo because the game can really slow down while the solver's running and it would look bad if we had to quit it for lack of a cancel button. maybe it's worth dropping a Thread.yield() somewhere in the solver code somewhere so the GUI gets some cpu time? @Gantonator

**Comment by userzjw** [comment]
Another tiny issue, when open the level editor, create a level without any walls like that
![image](https://cloud.githubusercontent.com/assets/18551565/14961835/bbf0217e-1093-11e6-8278-c399f83f01e5.png)
Then start the game, the map is still imported but the solver cannot solve the map
![image](https://cloud.githubusercontent.com/assets/18551565/14961877/09ec230a-1094-11e6-9a12-49eae8b9d833.png)

**Comment by tpict** [comment]
taking a look at that now - it should work, it'll just take a really long time

**Comment by Gantonator** [comment]
Yeah the solver can solve it eventually  It's just a huge level so bruteforce will take a long time.  On the solver window issues.  It is just in the background.  If you minimize all your windows you will see it.  It will also be behind the game executable so you have to move it out of the way.

**Comment by tpict** [comment]
on what version of windows is that josh?

**Comment by Gantonator** [comment]
10, but I assume rosie probably didn't minimize eclipse to look for the little JFrame window.  The problem is it doesn't get brought into focus when you select the game from the taskbar.  Its the same on windows 8.  If you press ctrl-D to you can see it.

**Comment by tpict** [comment]
apparently if you control click the java icon in the taskbar it should eventually pop up?

**Comment by Gantonator** [comment]
It's always there.  It just doesn't get brought to the front.  If you minimize the front-most window it will be there.  Remembering that it might still be behind the little game window too.

**Comment by tpict** [comment]
so long as it's not *disappearing* i think we can put that down to a fault of the windows window manager and not to our programming

**Comment by marcusjrc** [comment]
Guys, for the Pathnode class, should gCost be allowed negative integers?

**Comment by tpict** [comment]
it shouldn't, i'll throw an IllegalArgumentException in there

**Comment by tpict** [comment]
OK i've added that

there's probably a lot of cases where we permit null objects to be used in methods but i don't think anything disastrous could happen before java throws a NullPointerException
